### ISSUE TITLE
0.72 RN app not installed on IOS, parse and sematic issues.

### ISSUE THREAD

**Issue by arbuzaicer** [software-mansion/react-native-screens#1859]
### Description

RNScreens is crashing during installation for IOS platform with 6 issues.

Most of the issues are located in RNSearchBar.h, RNSearchBar.mm files.

<img width="721" alt="Знімок екрана 2023-08-03 о 8 15 20 пп" src="https://github.com/software-mansion/react-native-screens/assets/49945177/d709e8a3-08ef-4f08-92c0-6085f4454ff7">

<img width="1061" alt="Знімок екрана 2023-08-03 о 8 16 36 пп" src="https://github.com/software-mansion/react-native-screens/assets/49945177/ddeaa46f-25da-4e54-afab-ed2f8d503ac0">

<img width="1066" alt="Знімок екрана 2023-08-03 о 8 16 53 пп" src="https://github.com/software-mansion/react-native-screens/assets/49945177/3424b3a2-1795-4ec2-9c78-689ef69b85e7">

<img width="1050" alt="Знімок екрана 2023-08-03 о 8 17 06 пп" src="https://github.com/software-mansion/react-native-screens/assets/49945177/c7d8d473-951c-4c01-a6e1-d0c0181f2875">

### Steps to reproduce

1. install latest RN project `npx react-native@latest init AwesomeProject`
2. install rnscreens `yarn add react-native-screens`
3. Try to install the app for IOS through xCode.

### Snack or a link to a repository

https://github.com/

### Screens version

3.23.0

### React Native version

0.72.3

### Platforms

iOS

### JavaScript runtime

None

### Workflow

React Native (without Expo)

### Architecture

Fabric (New Architecture)

### Build type

None

### Device

iOS simulator

### Acknowledgements

Yes

**Comment by kkafar** [comment]
Hey @arbuzaicer, what version of iOS are you deploying for and what version of XCode do you use?

**Comment by SamiChab** [comment]
Same for me ! Deploying for iOS 12.4 with XCode 13.2.1

**Comment by kkafar** [comment]
Thanks @SamiChab,

I've broken this in #1843 by forgetting to add iOS API availability checks around method definitions and didn't spot the issue because I have newest SDK installed.

I've prepared the fix, will try to release minor version tomorrow.

**Comment by SamiChab** [comment]
Thank you @kkafar ✨

**Comment by kkafar** [comment]
Can please anyone affected confirm that fix in #1860 works? I have no disk space left to download & test older SDK 🙈 

You can check it out by installing `react-native-screens` directly from the branch in `package.json`: 

```json
"react-native-screens": "software-mansion/react-native-screens#@kkafar/build-problems-ios-16"
```

(clear cache & rebuild native code).

**Comment by arbuzaicer** [comment]
Thx 🙏 @kkafar

**Comment by kkafar** [comment]
I've released 3.24.0 with fix #1860, should be available to download straight from npm.

**Comment by arbuzaicer** [comment]
@kkafar successfully updated and installed. All good. thanks for the quick reaction :)
